src/main/java/ch/njol/skript/effects/EffOperations.java

```diff
@@ -154,8 +160,14 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 	@Override
 	protected void execute(Event event) {
 		Object rightObject = right.getSingle(event);
-		if (rightObject == null)
+		if (rightObject == null) {
+			error("Cannot operate with a null object.");
+			return;
+		}
+		if (left.isSingle() && left.getSingle(event) == null) {
+			error("Cannot operate on a null object.");
 			return;
+		}
```

Comment:
Shouldn't be doing errors here imo. Precedent is for effects acting on nothing to simply do nothing.